### PR TITLE
Prevent Leave role from being modified through role-add

### LINKED ISSUE #200: Role-add and Role-delete can remove all leave from system
`Leave` can be added as a role, which can lead to disaster when it is deleted - causing all leaves to be lost.

### PR DESCRIPTION
**Pull request by WangZijun97**
Resolves #200 by adding a new error message if a Leave is parsed in `role-add`.

### CODE REVIEW

File: src/main/java/mcscheduler/logic/commands/RoleAddCommand.java [hunk 1]
```diff
@@ -31,6 +33,9 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
         if (model.hasRole(toAdd)) {
+            if (toAdd.equals(new Leave())) {
```
Comment (wakululuu) [inline src/main/java/mcscheduler/logic/commands/RoleAddCommand.java]:
```suggestion
            if (Leave.isLeave(toAdd)) {
```

File: src/main/java/mcscheduler/logic/commands/RoleDeleteCommand.java [hunk 2]
```diff
@@ -49,6 +50,8 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Role roleToDelete = roleList.get(targetIndex.getZeroBased());
+        assert !roleToDelete.equals(new Leave());
```
Comment (wakululuu) [inline src/main/java/mcscheduler/logic/commands/RoleDeleteCommand.java]:
```suggestion
        assert !Leave.isLeave(roleToDelete);
```

**Review by wakululuu — APPROVED** [review]
LGTM other than a minor suggestion to use `Leave#isLeave` instead of `Leave#equals`

**Comment by WangZijun97** [comment]
Thanks, suggestion added.

**Pull request merged**